### ISSUE TITLE
Root catalog has invalid STAC object somewhere in the child & item link relation tree

### ISSUE THREAD

**Issue by philvarner** [jjrom/resto#356]
**Describe the bug**

Somewhere in the Catalog child & item link relation tree starting at https://tamn.snapplanet.io there is an invalid STAC object. 

Running this code, which traverses those links to retrieve items:

```python
import pystac
list(pystac.Catalog.from_file("https://tamn.snapplanet.io/").get_all_items())
```

results in `pystac.errors.STACTypeError: JSON does not represent a STAC object.`

There's not a lot of information from that error, so it might require stepping through in a debugger.

**Comment by jjrom** [comment]
Effectively corrected in commit https://github.com/jjrom/resto/commit/aae5294f50f80fa3e24d2999de2eaa224527f0e7

**Comment by philvarner** [comment]
I let the pystac team know about this. However, I can't reproduce it locally.

**Comment by jjrom** [comment]
@gadomski Thanks for adding context to the issue. You're right that resto did not encode the url path. This is corrected in #360 and tamn.snapplanet.io is updated accordingly
